### DISCUSSION TITLE
Helm chart?

### DISCUSSION THREAD

**Discussion by perfectra1n** [TriliumNext/Notes#40] [General]
Hey all,

I would love to help contribute in any way, the only caveat is that I'm trash with JS. I was thinking of creating a Helm chart though, so that anyone running a Kubernetes cluster could easily spin up Trilium. I don't believe that any tweaks are needed for the Dockerfile, but I wanted to gather y'all's feedback on creating a Helm Chart repo for Trilium :)

(I can write tons of Python/Bash/Docker/Kubernetes stuff though, so feel free to point me to anything that needs revising, I'll also write documentation for it :))

**Comment by perfectra1n** [comment]
Similar to Immich's Helm chart, I was going to use bjw-s's app-template (which I also used to write the Vikunja Helm chart) :)

**Comment by perfectra1n** [comment]
Would you be willing to create a repo for this @eliandoran? Maybe something generic like `trilium-helm-chart`? :)

(I've written the [Vikunja](https://kolaente.dev/vikunja/helm-chart/commits/branch/main) Helm chart already, and have quite a bit of experience with creating them)

Edit: oops, I just realized I can create repos too - so I guess the question is, are you alright with me creating the repo? I wouldn't want to freak anyone out lol

**Reply by eliandoran** [reply]
Hi @perfectra1n , I've created a repository for you: https://github.com/TriliumNext/helm-chart 

i've shortened the name to `helm-chart` since we already have Trilium in the org name and I like the look of it. :)

Good luck!

**Reply by perfectra1n** [reply]
Awesome, thanks! The name makes sense, I thought for sure that I wouldn't have perms to create a repo 😂 

If you're alright with it - I might need to ping you via email to help set up remote repositories like ArtifactHub (so that we're easily found and so that my dumb self doesn't own the organization, unless you'd want me to). These remote repositories require creating an account and then adding a sort of public key into the Helm Chart release so that it knows we're the rightful owners :)

**Reply by eliandoran** [reply]
Sure, feel free to drop me a line.
I'm OK with handling the accounts, to have everything centralized for now.

**Reply by perfectra1n** [reply]
Cool, sounds good! I'm glad you undestood my rambling :)

**Comment by perfectra1n** [comment]
Hey @eliandoran, I'm pretty close to being ready to release it to GitHub Pages (which full disclosure, I haven't had to set up before). I was hoping to have a release process that mirrored [Immich's](https://github.com/immich-app/immich-charts) pretty closely, and they use the URL `https://immich-app.github.io/immich-charts` to host the charts.  


[This](https://github.com/helm/chart-releaser-action) is the GitHub Action that I'm hoping to use to release it. I don't have access to the settings of the repository to meet the prerequisites [here](https://github.com/helm/chart-releaser-action?tab=readme-ov-file#pre-requisites). Would you be able to do Step 3 for me? I already created the `gh-pages` branch. A GitHub token will [need to be used](https://github.com/immich-app/immich-charts/blob/main/.github/workflows/release.yaml#L26) as well to create the release, does my account have privileges to create releases in the repository, and update the repository's GitHub Pages or would you want/need to create a limited scope token from your user, to do that?

**Comment by meichthys** [comment]
Moving to #340 and closing since a Helm chart is now available! 
https://github.com/TriliumNext/helm-charts
